src/users/users.repository.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable, NotFoundException } from '@nestjs/common'
+import { PrismaService } from 'src/prisma/prisma.service'
+import { UserEntity } from 'src/users/entities/user.entity'
+
+@Injectable()
+export class UsersRepository {
+    constructor(private prisma: PrismaService) {}
+
+    async getManyUsers(
+        page: number,
+        pageSize: number,
+        filter: Record<string, string>,
+    ): Promise<UserEntity[]> {
+        const users = await this.prisma.users.findMany({
+            where: filter,
+            take: pageSize,
+            skip: (page - 1) * pageSize,
+            include: {
+                Student: true,
+            },
+        })
+
+        return users
+    }
+
+    async getUserById(id: number): Promise<UserEntity> {
+        const user = await this.prisma.users.findUnique({
+            where: {
+                id,
+            },
+            include: {
+                Student: true,
```

Comment:
I think this way it work in some case also u can add params `returnStudent` is boolean to pass into function